### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by Robinlovelace**
- Layers: Refactor addDataSources to avoid duplicated code
- Layers: Refactor legend drawing
- Layers: Simplify style_head
- Layers: Start to simplify data zones check box
- Layers: In data zones, hoist definitions, so all logic now grouped

### CODE REVIEW

File: js/layer_control.js [hunk 1]
```diff
@@ -492,21 +492,21 @@ function switch_style(){
     // Manage placenames
     placenames();
     
-    // Sliders 
-    quietnessSlider.noUiSlider.on('update', function(){
+    // Sliders
+    sliders.quietness.noUiSlider.on('update', function(){
```
Comment (Robinlovelace) [inline js/layer_control.js]: Logical function name. There are many with identical content, could these be  be replaced with an iterator over each of the attributes? Great work in any case!

File: js/layer_control.js [hunk 2]
```diff
@@ -638,12 +638,12 @@ function switch_rnet() {
   var layerScenario = document.getElementById("rnet_scenario_input").value;
   var layerColour = document.getElementById("rnet_colour_input").value;
   var layerType = document.getElementById("rnet_type_input").value;
-  var sliderQuietness_min = Number(quietnessSlider.noUiSlider.get()[0]);
-  var sliderQuietness_max = Number(quietnessSlider.noUiSlider.get()[1]);
-  var sliderGradient_min = Number(gradientlider.noUiSlider.get()[0]);
-  var sliderGradient_max = Number(gradientlider.noUiSlider.get()[1]);
-  var sliderFlow_min = Number(cycleSlider.noUiSlider.get()[0]);
-  var sliderFlow_max = Number(cycleSlider.noUiSlider.get()[1]);
+  var sliderQuietness_min = Number(sliders.quietness.noUiSlider.get()[0]);
```
Comment (Robinlovelace) [inline js/layer_control.js]: Still seems quite hard-coded and procedural to my untrained eye. Main thing though: it works and is easy to maintain.

**Review by Robinlovelace — COMMENTED** [review]
Looking good.

File: js/layer_control.js (hunk 2 above)
Comment (mvl22) [reply js/layer_control.js]: Ditto; don't assume that anything you see currently is the final state - there are still days of work to be done. For instance that section you quoted was already changed in a subsequent commit, and there will be several further transformations.

File: js/layer_control.js (hunk 1 above)
Comment (mvl22) [reply js/layer_control.js]: Subsequent stages has now refactored these to a loop, and then auto-detects all cases to be converted to sliders based on the HTML, so there is not even now a list of the slider names defined in the logic at all:

https://github.com/nptscot/nptscot.github.io/blob/refactor/js/sliders.js

Any div with a `slider-styled` class will now become a slider, and what was previously hard-coded in the JS file is now set as attributes in an HTML `<datalist>`, e.g.:

https://github.com/nptscot/nptscot.github.io/blob/54528ae7fc3886d12a21677dcf3a881a910e67b1/index.html#L213-L224

Comment (Robinlovelace) [reply js/layer_control.js]: Great to hear! All about the input config, more data-driven approach.

Comment (mvl22) [reply js/layer_control.js]: Datalist is a standardised HTML5 tag nowadays, so is the most appropriate place to put these values. A standard HTML5 slider (which is too basic to use in this app, unfortunately) uses datalist, so this methodology is very close to the standard HTML approach.

**Pull request merged**